### ISSUE TITLE
Unable to run watchtower on ARM 64 bit architecture

### ISSUE THREAD

**Issue by scienmind** [containrrr/watchtower#779]
**Describe the bug**
Running Watchtower on ARM64 platform results in an error
```
The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
```

**To Reproduce**
```
user@host:~ $ docker run -d --name watchtower -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower:arm64v8-latest
Unable to find image 'containrrr/watchtower:arm64v8-latest' locally
arm64v8-latest: Pulling from containrrr/watchtower
3ed6cec7d4d1: Pull complete 
d127eddd71e3: Pull complete 
8a9c414386ab: Pull complete 
Digest: sha256:21d38836af48fdafc21a60f948613501917a1b740459a4a0531be5a65c7de3ca
Status: Downloaded newer image for containrrr/watchtower:arm64v8-latest
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
a817da09fb411ad7df92c0fbd661ff34269fffddd6bfc3648309487630d11abf
```

**Expected behavior**
Watchtower runs on ARM64 architecture without errors/warnings.

**Environment**
 - Platform
 - CPU Architecture:
 ```
user@host:~ $ lscpu
Architecture:        aarch64
Byte Order:          Little Endian
CPU(s):              4
On-line CPU(s) list: 0-3
Thread(s) per core:  1
Core(s) per socket:  4
Socket(s):           1
Vendor ID:           ARM
Model:               3
Model name:          Cortex-A72
Stepping:            r0p3
CPU max MHz:         1500.0000
CPU min MHz:         600.0000
BogoMIPS:            108.00
Flags:               fp asimd evtstrm crc32 cpuid
 ```

 - Docker version 
 ```
user@host:~ $ docker version
Client: Docker Engine - Community
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:18:36 2020
 OS/Arch:           linux/arm64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8891c58
  Built:            Mon Dec 28 16:16:39 2020
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
 ```

**Comment by simonaronsson** [comment]
Hey, thanks you for making us aware! Yeah, this is clearly wrong. We'll make sure it's corrected in the next version. Sorry for the inconvenience!

**Comment by scienmind** [comment]
Thanks for the quick response :) 

I also tried running armhf images, with the same result (`... image's platform (linux/amd64) does not match ...`), so the issue is probably not 64 bit specific, but relevant for all arm images.

BTW, in case it's helpful:
Portainer had a similar issue with their recent release: https://github.com/portainer/portainer/issues/4663
(The cause seemed to be that x86_64 bases image were mistakenly used for child arm ones.)

**Comment by piksel** [comment]
I can confirm that the manifest is **correct**:
```json
{
    "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
    "size": 948,
    "digest": "sha256:21d38836af48fdafc21a60f948613501917a1b740459a4a0531be5a65c7de3ca",
    "platform": {
        "architecture": "arm64",
        "os": "linux",
        "variant": "v8"
    }
}
```
but upon inspecting that image:

<details>
<summary>docker image inspect 3ed6cec7d4d1203e8e81f5c58898da0057d7ec3d12b9f2144b219dc657a27d5f</summary>

```json
[
    {
        "Id": "sha256:1bcdb0041be3f316427f20c1c4270104b534dd290f9f6a9318513a43e54fe690",
        "RepoTags": [
            "containrrr/watchtower:arm64v8-latest"
        ],
        "RepoDigests": [
            "containrrr/watchtower@sha256:21d38836af48fdafc21a60f948613501917a1b740459a4a0531be5a65c7de3ca"
        ],
        "Parent": "",
        "Comment": "",
        "Created": "2021-01-10T10:32:39.797226003Z",
        "Container": "b042a0848ba0411808bcdcc1147f828b018ee5f61540f70611a099eac43571eb",
        "ContainerConfig": {
            "Hostname": "b042a0848ba0",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "8080/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "#(nop) ",
                "ENTRYPOINT [\"/watchtower\"]"
            ],
            "Image": "sha256:f59c485ab833e906714857a64e1134529cf00da86405a1e86d0652a55d302bb1",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/watchtower"
            ],
            "OnBuild": null,
            "Labels": {
                "com.centurylinklabs.watchtower": "true"
            }
        },
        "DockerVersion": "17.09.0-ce",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "8080/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": null,
            "Image": "sha256:f59c485ab833e906714857a64e1134529cf00da86405a1e86d0652a55d302bb1",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/watchtower"
            ],
            "OnBuild": null,
            "Labels": {
                "com.centurylinklabs.watchtower": "true"
            }
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 14694803,
        "VirtualSize": 14694803,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/13ff016bf3471770091177c66bf50d102b7d9ed45c0746f4831e36318e75463b/diff:/var/lib/docker/overlay2/da660306d99a5aa28111b060c176c36a332c43eb88c5385ab20e258249171476/diff",
                "MergedDir": "/var/lib/docker/overlay2/a5d98c471d88eef2852e06ead1fed6c58220db344f64dc50e2a4b80d8db4a9ec/merged",
                "UpperDir": "/var/lib/docker/overlay2/a5d98c471d88eef2852e06ead1fed6c58220db344f64dc50e2a4b80d8db4a9ec/diff",
                "WorkDir": "/var/lib/docker/overlay2/a5d98c471d88eef2852e06ead1fed6c58220db344f64dc50e2a4b80d8db4a9ec/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:c1bc357da19de9c82758fa3f698e7330d2403024f255f0cd80aa77d5c5eddf2f",
                "sha256:1b4039eb5c1755b085335b60c9e1e8bbf91a72e6ada5a0033073597fe45b16c1",
                "sha256:6e5741b5d1f02171daf08a8fb91b8c15ef24eaca520dd2bf57e4975255828194"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
```

</details>

docker thinks it's **amd64**. I have no idea what is going on here...

Also, since the images are made from scratch, there is no base image.

**Comment by simonaronsson** [comment]
> I can confirm that the manifest is **correct**:
> 
> ```json
> {
>     "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>     "size": 948,
>     "digest": "sha256:21d38836af48fdafc21a60f948613501917a1b740459a4a0531be5a65c7de3ca",
>     "platform": {
>         "architecture": "arm64",
>         "os": "linux",
>         "variant": "v8"
>     }
> }
> ```
> 
> but upon inspecting that image:
> 
> docker image inspect 3ed6cec7d4d1203e8e81f5c58898da0057d7ec3d12b9f2144b219dc657a27d5f
> 
> docker thinks it's **amd64**. I have no idea what is going on here...
> 
> Also, since the images are made from scratch, there is no base image.

It could very well be a manifest level error or something borked in goreleaser. We'll have to investigate this a bit prior to releasing the next version, I think.

**Comment by Freekers** [comment]
I receive the same warning on my Synology DS218Play with Realtek RTD1296 cpu, but watchtower runs fine regardless. It doesn't matter if I'm using latest or the specific ARM tag, the warning always shows up.

```
processor       : 0
model name      : ARMv8 Processor rev 4 (v8l)
BogoMIPS        : 54.00
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 1
model name      : ARMv8 Processor rev 4 (v8l)
BogoMIPS        : 54.00
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 2
model name      : ARMv8 Processor rev 4 (v8l)
BogoMIPS        : 54.00
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 3
model name      : ARMv8 Processor rev 4 (v8l)
BogoMIPS        : 54.00
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4
```

```
Client: Docker Engine - Community
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:13:53 2020
 OS/Arch:           linux/arm64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8891c58
  Built:            Mon Dec 28 16:18:17 2020
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by Pterko** [comment]
Probably this issue is connected with https://github.com/docker/for-linux/issues/1170
As I understood, multi-arch images should work, but single-arch images with arm arhitecture are affected by this issue.

**Comment by piksel** [comment]
@Pterko, thanks for that link. It pointed me in the right direction. Apparently there is an ~~undocumented~~ `--platform` argument to `docker build` that you need to use with from-scratch images.

All prior watchtower images have been using `amd64` as the listed architecture in the images, but since the Docker daemon didn't care about that field until v20.10 no one noticed (and it didn't matter).

PR #811 should fix the issue.

**Edit**: Actually, the `--platform` argument *is* listed in the Docker documentation (but not in the CLI). Albeit with the only description being: "Set platform if server is multi-platform capable" and no further mention.

**Comment by simonaronsson** [comment]
This will be released in `v1.2.0`
